src/operation/decision_model.jl

```diff
@@ -280,7 +280,19 @@ end
 get_horizon(model::DecisionModel) = get_horizon(get_settings(model))
 
 """
-Implementation of build for any DecisionProblem
+Build the Decision Model based on the specified DecisionProblem.
+
+# Arguments
+
+  - `model::DecisionModel{<:DecisionProblem}`: DecisionModel object
+  - `output_dir::String`: Output directory for results
+
+# Accepted Key Words
```

Comment:
I don't think that there should be a separate section called "Accepted Key Words." You might check the latest docstring guidelines, but I suspect this should be under "Arguments".

Reply:
It seems that you are right given this: https://docs.julialang.org/en/v1/manual/documentation/

```
For simple functions, it is often clearer to mention the role of the arguments directly in the description of the function's purpose. An argument list would only repeat information already provided elsewhere. However, providing an argument list can be a good idea for complex functions with many arguments (in particular keyword arguments). In that case, insert it after the general description of the function, under an # Arguments header, with one - bullet for each argument. The list should mention the types and default values (if any) of the arguments:
```

On another note, do we keep the main method at the top of the docstring or should we just keep the list of methods at the end provided by DocStringExtensions?

Reply:
@jd-lara @daniel-thom I prefer to keep the general method on top of the dosctring instead of listing all the methods at the end based on docextensions:

```
@template DEFAULT = """
                    $(TYPEDSIGNATURES)
                    $(DOCSTRING)
                    $(METHODLIST)
                    """
```

Something that happens with this is that it does not look clean, like this:

![image](https://user-images.githubusercontent.com/24548207/232911924-9963fdec-876c-4c4f-872f-9a2273aa7723.png)

My vote is to remove the MethodList, and manually set-up the main method at the beginning. At least that is how they handle in JuMP

Reply:
Yeah, I removed the METHODLIST, it does not look good and is not being used in PSY. With that, all methods will autogenerate the typed signature.

However, for the types, such as DeviceModel or DecisionModel it will not work.

So what about this plan:
 - Remove METHODLIST
 - Use DocStringExtensions for typed signatures for methods (such as `build!` or `solve!`)
 - Manually create the signature for relevant Types (such as `DeviceModel` or `DecisionModel`).

What do you think?

Reply:
removing Method list is ok. We will end up with an explosion of methods otherwise